### ISSUE TITLE
[NFR]: pre-release on pecl

### ISSUE THREAD

**Issue by remicollet** [phalcon/cphalcon#15421]
Is there any plan to publish pre-release on PECL ?  (such as 5.0.0-alpha1)

If stability is properly set to "alpha/beta" in package.xml it won't be used by default unless people especially ask for them
and it will provide more visibility to dev progress toward PHP 8 support

**Comment by Jeckerson** [comment]
Hello

We will need to wait @ruudboon as he is responsible to upload to pecl. 

P.S. We wait for x2 years to get access to pecl :joy:

**Comment by remicollet** [comment]
ok, so this is a global issue (4.1.1 also missing on pecl)

**Comment by remicollet** [comment]
If you cannot upload to pecl, perhaps you can attached the pecl archive in the release assets ?

**Comment by ruudboon** [comment]
We had an issue with the 4.1.1 and that blocked us from pushing it to Pecl.  Pushing a new release now.  
Would love to do a alpha release on Pecl as well and let's see if we can the pecl archive automatically upon release.

**Comment by StudioMaX** [comment]
Please also upload 4.1.2 to the packagecloud repository if it is already stable.

**Comment by Jeckerson** [comment]
@StudioMaX Uploaded. Please try to install.

**Comment by ruudboon** [comment]
Currently the version number is blocking us releasing the alpha on Pecl. In source our version number is `5.0.0-alpha.2` in the xml we have `5.0.0-alpha.2` versions must match to release the alpha. Unfortunately it looks like Pecl isn't accepting dashes in version numbers.

**Comment by remicollet** [comment]
> Currently the version number is blocking us releasing the alpha on Pecl. In source our version number is 5.0.0-alpha.2 in the xml we have 5.0.0-alpha.2 

AFAIK, the check is only a warning, not a blocker

> Unfortunately it looks like Pecl isn't accepting hyphen in version numbers.

hyphen are only used for "-dev", not for alpha, beta, RC

**Comment by ruudboon** [comment]
> AFAIK, the check is only a warning, not a blocker

It is a blocker. Won't be able to continue the upload. I just tested this.

> hyphen are only used for "-dev", not for alpha, beta, RC
I see we use it here to indicate a 'special' version: https://github.com/phalcon/cphalcon/blob/198e45905131dabb6020097ca7e2546128c1f75c/phalcon/Support/Version.zep#L134
Not remembering from the top of my head if Zephir is getting the version from here or we have a different mechanism that uses the same approach.

I think we should comply with the php standard for versioning. Found some stuff here: https://pear.php.net/manual/en/pyrus.commands.pickle.php

From what I found I think we should name alpha/beta/rc like `5.0.0a2` beta like `5.0.0b1` and RC like `5.0.0RC1` .

**Comment by remicollet** [comment]
> From what I found I think we should name alpha/beta/rc like 5.0.0a2 beta like 5.0.0b1 and RC like 5.0.0RC1 .

Rather 5.0.0alpha2, 5.0.0beta1 and 5.0.0RC1

**Comment by ruudboon** [comment]
> Rather 5.0.0alpha2, 5.0.0beta1 and 5.0.0RC1
I prefer this notation as well. Just didn't found it in the documentation. I see the Derick is doing this as well https://pecl.php.net/package/xdebug so I think we should be fine on this approach.

**Comment by remicollet** [comment]
About alpha3, I notice that archive generated from the git clone (`phalcon-5.0.0alpha3.tgz`)  is very different than the asset attached to the tag (the `phalcon-pecl.tgz` in the `phalcon-pecl.zip`), main being the used zephir version

```
$ diff -ru hub/php_phalcon.h git/php_phalcon.h
--- hub/php_phalcon.h   2021-06-30 18:46:16.000000000 +0200
+++ git/php_phalcon.h   2021-07-01 07:19:46.000000000 +0200
@@ -106,7 +106,7 @@
 #define PHP_PHALCON_VERSION     "5.0.0alpha3"
 #define PHP_PHALCON_EXTNAME     "phalcon"
 #define PHP_PHALCON_AUTHOR      "Phalcon Team and contributors"
-#define PHP_PHALCON_ZEPVERSION  "0.13.4-8d3141c8"
+#define PHP_PHALCON_ZEPVERSION  "0.13.5-1232784"
 #define PHP_PHALCON_DESCRIPTION "Phalcon is a full stack PHP framework, delivered as a PHP extension, offering lower resource consumption and high performance."
 
 typedef struct _zephir_struct_db { 
```

Which is the proper one ?

**Comment by ruudboon** [comment]
Hmm we need to check this before pushing it into pecl. I will have a look today

**Comment by remicollet** [comment]
Notice, for the RPM build in my repo, I used the one generated from the git clone (`git checkout 5.0.0alpha3 ; pecl package`)

**Comment by remicollet** [comment]
Looks like the tag is set in master branch (44aa04c9871b4d9f14c8815b3b46df1817316074) instead of the 5.0.x branch...
And the asset miss the 2 last commits (so at c1d3f217d65cd552fbc2be6e6c9f27a3ebfd2312)

**Comment by remicollet** [comment]
So the one released on pecl uses 0.13.4 and not 1.13.5, is that expected ?

**Comment by ruudboon** [comment]
> So the one released on pecl uses 0.13.4 and not 1.13.5, is that expected ?
@remicollet Yes. Mainly stubs are fixed in the 13.5 otherwise we had to wait for the next release. Dropped a pull to sync everything and hopefully we can drop the next alpha soon. Closing this issue since we are having a release now. Every next release will be published as well.
